plugins/woocommerce/includes/class-wc-countries.php

```diff
@@ -474,6 +474,11 @@ public function country_dropdown_options( $selected_country = '', $selected_stat
 			foreach ( $this->countries as $key => $value ) {
 				$states = $this->get_states( $key );
 				if ( $states ) {
+					// Maybe default the selected state as the first one.
+					if ( '*' === $selected_state ) {
+						$selected_state = key( $states );
```

Comment:
We only want to select a default state in countries that have states.

```suggestion
						$selected_state = key( $states ) ?? '*';
```